### ISSUE TITLE
Question about training label

### ISSUE THREAD

**Issue by zfang399** [facebookresearch/taskmodularnets#5]
Hi!

This work is really interesting! Thanks for sharing the codes. I tried running the training script and it looks to be working.

I have a question about the training label: it seems that during training the label is always set to all zeros and fed into the cross entropy loss:
`labels = torch.zeros_like(sampled_attrs[:, 0]).long()`

Could you help me with understanding this? Thanks!

**Comment by senthilps8** [comment]
Thanks for your interest. 
We use zeros as the labels because the first class sample is the positive ground truth class and the rest are the sampled negative classes as can be seen in this line:
https://github.com/facebookresearch/taskmodularnets/blob/7e27bc444f03615cc9193edb275728282352052d/archs/models.py#L384

**Comment by zfang399** [comment]
I see! Thanks a lot!
